apps/comps/e2e/utils/rpc-client-helpers.ts

```diff
@@ -80,3 +102,428 @@ export async function createTestRpcClient(
+ */
+export async function createTestCompetition({
+  adminRpcClient,
+  name,
+  description,
+  sandboxMode,
+  externalUrl,
+  imageUrl,
+  type,
+  tradingType,
+  startDate,
+  endDate,
+  boostStartDate,
+  boostEndDate,
+  joinStartDate,
+  joinEndDate,
+  maxParticipants,
+  tradingConstraints,
+  rewardsIneligible,
+  paperTradingInitialBalances,
+}: {
+  adminRpcClient: RouterClient<typeof adminRouter>;
+} & Partial<z.infer<typeof AdminCreateCompetitionSchema>>) {
+  const competitionName = name || `Test competition ${Date.now()}`;
+  const result = await adminRpcClient.competitions.create({
+    name: competitionName,
+    description:
+      description || `Test competition description for ${competitionName}`,
+    tradingType,
+    sandboxMode,
+    externalUrl,
+    imageUrl,
+    type,
+    startDate: startDate instanceof Date ? startDate.toISOString() : startDate,
+    endDate: endDate instanceof Date ? endDate.toISOString() : endDate,
+    boostStartDate:
+      boostStartDate instanceof Date
+        ? boostStartDate.toISOString()
+        : boostStartDate,
+    boostEndDate:
+      boostEndDate instanceof Date ? boostEndDate.toISOString() : boostEndDate,
+    joinStartDate:
+      joinStartDate instanceof Date
+        ? joinStartDate.toISOString()
+        : joinStartDate,
+    joinEndDate:
+      joinEndDate instanceof Date ? joinEndDate.toISOString() : joinEndDate,
+    maxParticipants,
+    tradingConstraints,
+    rewardsIneligible,
+    arenaId: "default-paper-arena",
+    paperTradingInitialBalances:
+      paperTradingInitialBalances || defaultPaperTradingInitialBalances(),
+  });
+
+  if (!result.success) {
+    throw new Error("Failed to create competition");
+  }
+
+  return result;
+}
+
+/**
+ * Start a perpetual futures test competition (creates and immediately starts it)
+ */
+export async function startPerpsTestCompetition({
+  adminRpcClient,
+  name,
+  agentIds,
+  sandboxMode,
+  externalUrl,
+  imageUrl,
+  tradingConstraints,
+  rewards,
+  evaluationMetric,
+  perpsProvider = {
+    provider: "symphony" as const,
+    initialCapital: 500,
+    selfFundingThreshold: 0,
+    apiUrl: "http://localhost:4567", // Point to mock server by default
+  },
+}: {
+  adminRpcClient: RouterClient<typeof adminRouter>;
+  name?: string;
+  agentIds?: string[];
+  sandboxMode?: boolean;
+  externalUrl?: string;
+  imageUrl?: string;
+  tradingConstraints?: z.infer<
+    typeof AdminStartCompetitionSchema
+  >["tradingConstraints"];
+  rewards?: Record<number, number>;
+  evaluationMetric?: "calmar_ratio" | "sortino_ratio" | "simple_return";
+  perpsProvider?: {
+    provider: "symphony" | "hyperliquid";
+    initialCapital?: number;
+    selfFundingThreshold?: number;
+    minFundingThreshold?: number;
+    apiUrl?: string;
+  };
+}) {
+  const competitionName = name || `Perps Test Competition ${Date.now()}`;
+  const result = await adminRpcClient.competitions.start({
```

Comment:
Just a note for if/when we re-write the competition lifecycle functions... I hate that you can provide all these parameters when starting a comp. It would be cleaner to let the `createComp` and `updateComp` functions accept all the parameters to configure the comp, and `startComp` take maybe 0 parameters.